app/controllers/visualizations_controller.rb

```diff
@@ -246,16 +246,21 @@ def displayVis
     num_field = ''
     # push real fields to temp variable
     @project.fields.sort_by(&:index).each do |field|
-      data_fields.push(typeID: field.field_type, unitName: field.unit, fieldID: field.id, fieldName: field.name)
+      data_fields.push(typeID: field.field_type, unitName: field.unit, fieldID: "#{field.id}", fieldName: field.name)
       # check if location and timestamp fields exist
       lat = field.id.to_s if field.field_type == 4
       time_field = field.id.to_s if field.field_type == 1
       num_field = field.id.to_s if field.field_type == 2
     end
 
+    # push formula fields to temp variable
+    @project.formula_fields.sort_by(&:index).each do |field|
+      data_fields.push(typeID: field.field_type, unitName: field.unit, fieldID: "f#{field.id}", fieldName: field.name, formula: true)
+    end
+
     has_pics = false
     has_loc_data = false
-    has_time_data = false
```

Comment:
It's several lines north of where I call it.  It's not shown in the diff for some reason.